src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/RouteTokenTransformerConvention.cs

```diff
@@ -8,7 +8,7 @@ namespace Microsoft.AspNetCore.Mvc.ApplicationModels
 {
     /// <summary>
     /// An <see cref="IActionModelConvention"/> that sets attribute routing token replacement
-    /// to use the specified <see cref="IOutboundParameterTransformer"/> on <see cref="ActionModel"/> selectors.
+    /// to use the specified <see cref="IOutboundParameterTransformer"/> on <see cref="ActionModel"/>.
     /// </summary>
```

Comment:
Worth explicitly mentioning in docs that this doesn't apply to pages and vice-versa?

Reply:
I think so.

The way I had to implement the route transform for pages makes me wonder if `RouteTokenTransformerConvention`'s behavior should change. Right now it just transforms area/controller/action route tokens. We could make it transform all the route's literal segments just like pages.

What do you think?

Reply:
ehhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

I don't think so. I think pages are special. It's a case where you have literal segments that you didn't write. In all of the other cases, if you have literals, YOU WROTE IT.

Reply:
Yeah I agree.

It is a little murky with the page route transformer because it does run the transform on part of the page route that you define: https://github.com/aspnet/Mvc/pull/8541/files#diff-f9a10458395befdb555adcbd76d85ba1R1

By the time the transformer runs, the directory and file name has already been merged with this route fragment.